crates/matrix-sdk/src/encryption/backups/mod.rs

```diff
@@ -387,7 +397,38 @@ impl Backups {
     /// This method will request info about the current backup from the
     /// homeserver and if a backup exits return `true`, otherwise `false`.
     pub async fn exists_on_server(&self) -> Result<bool, Error> {
-        Ok(self.get_current_version().await?.is_some())
+        let exists_on_server = self.get_current_version().await?.is_some();
+        *self.cached_exists_on_server.write().unwrap() = Some(exists_on_server);
+        Ok(exists_on_server)
+    }
+
+    /// Does a backup exist on the server?
+    ///
+    /// This method is identical to [`Self::exists_on_server`] except that we
+    /// cache the latest answer in memory and only empty the cache if the local
+    /// device adds or deletes a backup itself.
+    ///
+    /// Do not use this method if you need an accurate answer about whether a
+    /// backup exists - instead use [`Self::exists_on_server`]. This method is
+    /// useful when performance is more important than guaranteed accuracy,
+    /// such as when classifying UTDs.
+    pub async fn fast_exists_on_server(&self) -> Result<bool, Error> {
```

Comment:
Not a huge fan of the name, I wonder if we could do it the opposite, rename this to `exists_on_server()`, and have the existing one renamed to `fetch_exists_on_server()` or something that implies a request is running?

Reply:
Done in 88afa6bff9 and f5f456398a